cmd/integrationArtifactResource.go

```diff
@@ -0,0 +1,229 @@
+	if fileExt == "" {
+		return errors.New("invalid file extension in resource file")
+	}
+	deleteIflowResourceStatusURL := fmt.Sprintf("%s/api/v1/IntegrationDesigntimeArtifacts(Id='%s',Version='%s')/$links/Resources(Name='%s',ResourceType='%s')", apiHost, config.IntegrationFlowID, "Active", fileName, fileExt)
+	deleteIflowResourceStatusResp, httpErr := httpClient.SendRequest(httpMethod, deleteIflowResourceStatusURL, nil, header, nil)
+
+	if deleteIflowResourceStatusResp != nil && deleteIflowResourceStatusResp.Body != nil {
+		defer deleteIflowResourceStatusResp.Body.Close()
+	}
+
+	if deleteIflowResourceStatusResp == nil {
+		return errors.Errorf("did not retrieve a HTTP response: %v", httpErr)
+	}
+
+	if deleteIflowResourceStatusResp.StatusCode == http.StatusOK {
+		log.Entry().
+			WithField("IntegrationFlowID", config.IntegrationFlowID).
+			Info("Successfully updated integration flow artefact in CPI designtime")
+		return nil
+	}
+	if httpErr != nil {
+		responseBody, readErr := ioutil.ReadAll(deleteIflowResourceStatusResp.Body)
+		if readErr != nil {
+			return errors.Wrapf(readErr, "HTTP response body could not be read, Response status code: %v", deleteIflowResourceStatusResp.StatusCode)
+		}
+		log.Entry().Errorf("a HTTP error occurred! Response body: %v, Response status code: %v", responseBody, deleteIflowResourceStatusResp.StatusCode)
+		return errors.Wrapf(httpErr, "HTTP %v request to %v failed with error: %v", httpMethod, deleteIflowResourceStatusURL, string(responseBody))
+	}
+	return errors.Errorf("Failed to update Integration Flow artefact, Response Status code: %v", deleteIflowResourceStatusResp.StatusCode)
+}
+
+//GetJSONPayload -return http payload as byte array
+func GetJSONPayload(config *integrationArtifactResourceOptions, mode string, fileUtils piperutils.FileUtils) (*bytes.Buffer, error) {
+	fileContent, readError := fileUtils.FileRead(config.ResourcePath)
+	if readError != nil {
+		return nil, errors.Wrapf(readError, "Error reading file")
+	}
+	fileName := filepath.Base(config.ResourcePath)
+	jsonObj := gabs.New()
+	if mode == "create" {
+		jsonObj.Set(fileName, "Name")
+		jsonObj.Set(GetResourceFileExtension(fileName), "ResourceType")
+		jsonObj.Set(b64.StdEncoding.EncodeToString(fileContent), "ResourceContent")
+	} else if mode == "update" {
+		jsonObj.Set(b64.StdEncoding.EncodeToString(fileContent), "ResourceContent")
+	} else {
+		return nil, fmt.Errorf("Unkown node: '%s'", mode)
+	}
+
+	jsonBody, jsonErr := json.Marshal(jsonObj)
+
+	if jsonErr != nil {
+		return nil, errors.Wrapf(jsonErr, "json payload is invalid for integration flow artifact %q", config.IntegrationFlowID)
+	}
+	return bytes.NewBuffer(jsonBody), nil
+}
+
+//GetResourceFileExtension -return resource file extension
+func GetResourceFileExtension(filename string) string {
+	fileExtension := filepath.Ext(filename)
+	switch fileExtension {
+	case ".xsl":
+		return "xslt"
+	case ".gsh":
+		return "groovy"
+	case ".groovy":
+		return "groovy"
```

Comment:
As the return value is the same I would suggest merging the cases like so:
```
	case ".gsh", ".groovy":
		return "groovy"
```

Reply:
done